book_programs/chapter_2/bertrands_paradox.py

```diff
@@ -0,0 +1,191 @@
+#   Usage: python3 bertrands_paradox.py <NCHORDS_TO_PLOT> <nchords> 
+#   NCHORDS_TO_PLOT:  The number of chords to plot
+#   Don't plot more than 1000 number of chords because they overlap too much
+#   and obscure the point we're trying to make.
+#   nchords:  The sample size of chords to do the statistics
+
+# Example Usage: python3 bertrands_paradox.py 1000 10000
+
+import numpy as np
+import sys
+import matplotlib.pyplot as plt
+from matplotlib.patches import Circle
+from matplotlib.lines import Line2D
+
+args = sys.argv
+
+#####################
+# User Input Handling
+#####################
+bad_input = False
+
+# Less than 2 arguments
+if len(args) < 2:
+    bad_input = True
+    print("ERROR: Insufficient arguments")
+
+# Try to convert arguments into integers
+try:
+    nchords = int(args[1])
+except IndexError:
+    pass
+except ValueError:
+    bad_input = True
+    print("ERROR: Could not convert 'Sample size of chords' to an integer")
+
+try:
+    NCHORDS_TO_PLOT = int(args[2])
+except IndexError:
+    pass
+except ValueError:
+    bad_input = True
+    print("ERROR: Could not convert 'No. of chords to plot' to an integer")
+
+# Argument n is negtive
+if 'nchords' in globals() and nchords < 0:
+    bad_input = True
+    print("ERROR: Sample size of chords cannot be negative")
+
+# Argument m is negtive
+if 'NCHORDS_TO_PLOT' in globals() and NCHORDS_TO_PLOT < 0:
+    bad_input = True
+    print("ERROR: No. of cords to plot cannot be negative")
+
+# Print usage help when arguments are bad
+if bad_input:
+    print("\nUsage: python3 bertrands_paradox.py <NCHORDS_TO_PLOT> <nchords>")
+    print("  NCHORDS_TO_PLOT is the number of chords to plot")
+    print("  nchords is the sample size of chords to do the statistics")
+    print("    Don't plot more than 1000 number of chords because they overlap too much")
+    print("    and obscure the point we're trying to make.")
+    print("\nExample Usage: python3 bertrands_paradox.py 1000 10000")
+    sys.exit(1)
+
+######################
+# Bertrand's Paradox #
+######################
+
+TAU = 2 * np.pi
+
+# Fractional RGB values for light grey.
+GREY = (0.2,0.2,0.2)
+# The circle radius. Doesn't matter what it is.
+r = 1
+# The critical side length of the equilateral triangle inscribed in the circle.
+# We are testing if a chord is longer than this length.
+tlen = r * np.sqrt(3)
+
+def setup_axes():
+    """Set up the two Axes with the circle and correct limits, aspect."""
+
+    fig, axes = plt.subplots(nrows=1, ncols=2, subplot_kw={'aspect': 'equal'})
+    for ax in axes:
+        circle = Circle((0,0), r, facecolor='none')

```

Comment:
could you please add an outline to the circle?

Reply:
Added a black outline
![Figure_1](https://user-images.githubusercontent.com/56556436/96220971-5e3dde00-0fa7-11eb-8b87-32627be893a3.png)